python/lsst/db/utils.py

```diff
@@ -397,46 +397,52 @@ def loadSqlScript(conn, script, dbName=None):
                        database used by connection/engine.
     """
 
-    # check file, if it has 'read' attribute assume it's a file object
-    if not hasattr(script, 'read'):
-        script = open(script)
-
     url = conn.engine.url
     if url.get_backend_name() == "mysql":
 
+        # check file, if it has 'read' attribute assume it's a file object
+        cleanup = None
+        if not hasattr(script, 'read'):
+            script = open(script)
+            cleanup = script.close
+
         # make tmpfile to store credentials and options
         fd, fname = tempfile.mkstemp(text=True)
```

Comment:
Ok. You win. I think the tail is wagging the dog a bit here. I'm not sure I understand why a special low level file descriptor is needed here rather than a boring Python-level file handle. There is no comment in the code explaining why that's necessary and it looks like the file is opened, text is written and the file is closed.